src/jit/morph.cpp

```diff
@@ -9469,12 +9469,7 @@ GenTree* Compiler::fgMorphInitBlock(GenTree* tree)
         }
         else
         {
-            if (dest->OperIs(GT_IND))
-            {
-                assert(dest->TypeGet() == TYP_STRUCT);
```

Comment:
This code is not really related to this fix, but is simply wrong, and obviously is never actually hit, since the sense is backward (it should be `dest->TypeGet() == TYP_STRUCT`, but in fact we should never actually see a `GT_IND` here.